internal/service/service.go

```diff
@@ -77,6 +77,36 @@ func buildURL(apiURL, orgID, format string) string {
 	)
 }
 
+func errorFromResponse(resp *http.Response, orgID string) error {
+	err := fmt.Errorf("could not convert to SBOM (status: %s)", resp.Status)
+	switch resp.StatusCode {
+	case http.StatusBadRequest:
+		return extension_errors.New(
+			err,
+			"SBOM generation failed due to bad input arguments. "+
+				"Please make sure you are using the latest version of the Snyk CLI.",
+		)
+	case http.StatusUnauthorized:
+		return extension_errors.New(
+			err,
+			"Snyk failed to authenticate you based on on you API token. "+
+				"Please ensure that you have authenticated by running `snyk auth`.",
+		)
+	case http.StatusForbidden:
+		return extension_errors.New(
+			err,
+			fmt.Sprintf(
+				"Your account is not authorized to perform this action. "+
```

Comment:
we dont take into account the error messages from the hidden endpoint do we?

Reply:
As of now, no we don’t. But I’m open to suggestions here. In a previous iteration the status code was part of the message (`Your account is not authorized to perform this action. (Status: 4xx)`) but that made it look cryptic to me.

Reply:
In that we should consider removing from the error catalog any errors only used by the hidden endpoint. [For example](https://github.com/snyk/sbom-export/blob/3fa03520a4c88e0a99dc52764eff8676b1a215e2/internal/hidden/2022-03-31/handler.go#L23)
Although, if it did use the error catalog I could see it being useful for the user to get a link to retrieve their API token; my understanding thats not possible yet for the cli.

Reply:
Yes, the way it works right now is that we return a Golang `error`, which basically only has a string value which will be displayed to the user. This is not ideal for many reasons, but most of all that you cannot make a distinction between a cryptic message that would help with debugging, and a user-friendly one that would read nicely and be helpful to our end users. Right now we are prioritizing the user-friendliness and disregarding the internal debugging stuff. This is a temporary solution until the error catalog makes it into the CLI, which should hopefully happen soon. We can then revise all of this and introduce error codes etc. as well.